x-pack/plugins/ml/public/application/notifications/components/ml_entity_selector.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useEffect, useState, useMemo } from 'react';
+import {
+  EuiComboBox,
+  type EuiComboBoxOptionOption,
+  type EuiComboBoxProps,
+  euiPaletteColorBlindBehindText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { countBy } from 'lodash';
+import { useMlApiContext } from '../../contexts/kibana';
+
+type EntityType = 'anomaly_detector' | 'data_frame_analytics' | 'trained_models';
```

Comment:
In other places in the notifications page `inference` is used.
I wonder if we're introducing confusion by using the label `Trained models` in this menu, even though it's probably clearer and we used `Trained model` everywhere else in the UI.
It's a shame the notifications use `inference` as the type.

Not asking for a change here, just thinking out loud.

Reply:
Agree, it's indeed confusing. Everywhere in the UI code, I refer to this as `trained models`. 
On the data level for the `.ml-notifications*` it's `inference`, if I'm not mistaken it's the only exception in the user-facing labels so far.